src/hashtable.c

```diff
@@ -319,8 +320,10 @@ static_assert(sizeof(hashtableIterator) >= sizeof(iter),
 /* Position, used by some hashtable functions such as two-phase insert and delete. */
 typedef struct {
     bucket *bucket;
-    uint16_t pos_in_bucket;
-    uint16_t table_index;
+    uint32_t *current_epoch; /* Pointer to the epoch of the hashtable. */
```

Comment:
It's interesting that the `position` doesn't include a reference to the `hashtable`.  So there's no mechanism to validate that a given position is actually associated with a given hashtable.

So, I see what you're doing here, keeping a reference to the epoch so that you can compare.  But wouldn't it be more useful to keep a reference to the hashtable instead?